src/java/org/apache/cassandra/metrics/DecayingEstimatedHistogramReservoir.java

```diff
@@ -414,19 +414,64 @@ public void rebase(EstimatedHistogramReservoirSnapshot snapshot)
             }
         }
 
-        this.decayLandmark = snapshot.snapshotLandmark;
+        DecayingBuckets newDecayingBuckets = new DecayingBuckets(snapshot.snapshotLandmark);
         for (int i = 0; i < size(); i++)
         {
             // set rebased values in the first stripe and clear out all other data
-            decayingBuckets.set(stripedIndex(i, 0), snapshot.decayingBuckets[i]);
+            newDecayingBuckets.buckets.set(stripedIndex(i, 0), snapshot.decayingBuckets[i]);
```

Comment:
Is rebase a frequent operation? If so, maybe instead of updating a new atomic long array, create just a long array, and then create atomic long array from that?

Reply:
Currently, this happens only on-demand during child-metrics removal.
The only realistic scenario I know of is changing table-related schema. Table-level latency metrics are children of global and keyspace latency metrics. Whenever the table is reloaded or removed the parent histograms get `rebased`.